mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -362,6 +381,16 @@ class RFormulaModel private[feature](
         case _ => true
       }
       StructType(withFeatures.fields :+ StructField($(labelCol), DoubleType, nullable))
+    } else if (resolvedFormula.evalExprs.contains(resolvedFormula.label)) {
+      val spark = SparkSession.builder().getOrCreate()
+      val dummyRDD = spark.sparkContext.parallelize(Seq(Row.empty))
+      val dummyDF = spark.createDataFrame(dummyRDD, schema)
+        .withColumn(resolvedFormula.label, expr(resolvedFormula.label))
+      val nullable = dummyDF.schema(resolvedFormula.label).dataType match {
+        case _: NumericType | BooleanType => false
```

Comment:
I think it can just be `case NumericType | BooleanType`?

Reply:
I thought so too! Then I got this error;

```
[error] .../spark/mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala:390: pattern type is incompatible with expected type;
[error]  found   : org.apache.spark.sql.types.NumericType.type
[error]  required: org.apache.spark.sql.types.DataType
[error] Note: if you intended to match against the class, try `case _: NumericType`
[error]         case NumericType | BooleanType => false


```